pkg/koordlet/metricsadvisor/collectors/performance/performance_collector_linux.go

```diff
@@ -191,9 +222,11 @@ func (p *performanceCollector) profilePerfOnSingleContainer(status *corev1.Conta
 
 	cpiMetrics = append(cpiMetrics, cpiCycle, cpiInstruction)
 
-	err1 := collectorOnSingleContainer.CleanUp()
-	if err1 != nil {
-		klog.Errorf("collectorOnSingleContainer cleanup err : %v", err1)
+	if !features.DefaultKoordletFeatureGate.Enabled(features.Libpfm4) {
```

Comment:
the CleanUp can be done with defer or just after GetContainerCyclesAndInstructions.